relay-server/src/service.rs

```diff
@@ -200,7 +202,7 @@ impl ServiceState {
         );
         ProjectCacheService::new(
             config.clone(),
-            buffer_guard.clone(),
+            memory_stat.init_checker(config.clone()),
```

Comment:
nit: Does this method have any advantage over  a constructor like `MemoryChecker::new(memory_stat.clone(), config.clone())`?

Reply:
I also debated about this, I might switch to the constructor like you showed to decouple even more the `MemoryStat` and `MemoryChecker`.